### PR TITLE
DEV: Make leaderboard respect `prioritize_usernames_in_ux`

### PR DESCRIPTION
**Pull request by rngus2344**

### CODE REVIEW

File: assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs [hunk 1]
```diff
@@ -9,7 +9,13 @@
...
   >
     {{avatar rank imageSize="large"}}
-    <span class="user__name">{{rank.username}}</span>
+    <span class="user__name">
+      {{#if this.siteSettings.prioritize_username_in_ux}}
+        {{rank.username}}
+      {{else}}
+        {{rank.name}}
+      {{/if}}
+    </span>
```
Comment (janzenisaac) [inline assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs:18]: Can we have a test for this logic to protect from regressions?

**Review by janzenisaac — CHANGES_REQUESTED** [review]
Looks good, thank you for the PR!

Can you update 

https://github.com/discourse/discourse-gamification/blob/7fbc2d072de3a7ae104875cd0edfd38445088e88/assets/javascripts/discourse/templates/components/minimal-gamification-leaderboard-row.hbs#L17

as well providing screenshots of the changes? 🎉

File: assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs (hunk 1 above)
Comment (Chanvannak12) [reply assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs:18]: A5FABF42-2F80-4F7E-BD28-F034D1023836

**Comment by rngus2344** [comment]
> We will need to update the `minimal-leaderboard`, you can see it here
> 
Quick question, when, where, or how do I see this `minimal-leaderboard`?

**Comment by rngus2344** [comment]
> In addition to this, could you provide screenshots of the changes? 🎉
Here are some screenshots of the changes! :
Screenshot 2023-05-25 at 12 30 47 PM
Screenshot 2023-05-25 at 12 30 18 PM

File: assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs (hunk 1 above)
Comment (rngus2344) [reply assets/javascripts/discourse/templates/components/gamification-leaderboard-row.hbs:18]: I just added some tests. Please let me know if there are any issues!

File: test/javascripts/components/gamification-leaderboard-row-test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+function displayName() {
+  return query(".user__name").innerText.trim();
+}
+
+module(
+  "Discourse Gamification | Component | gamification-leaderboard-row",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    test("name", async function (assert) {
```
Comment (janzenisaac) [inline test/javascripts/components/gamification-leaderboard-row-test.js:16]: Nitpick, but let's use a more clear description like `Display name prioritizes name` and `Display name prioritizes username`

**Comment by janzenisaac** [comment]
@rngus2344 small nitpick, and just need to cleanup the linting errors then we should be good to go!

File: test/javascripts/components/gamification-leaderboard-row-test.js (hunk 2 above)
Comment (rngus2344) [reply test/javascripts/components/gamification-leaderboard-row-test.js:16]: Yessir

**Comment by rngus2344** [comment]
> @rngus2344 small nitpick, and just need to cleanup the linting errors then we should be good to go!

It should be good now! Let me know if there is anything else!

**Pull request merged**